src/chainstate/burn/operations/leader_block_commit.rs

```diff
@@ -679,17 +679,43 @@ impl LeaderBlockCommitOp {
             //  is not valid, but we should allow this UTXO to "chain" to valid
             //  UTXOs to allow the miner windowing to work in the face of missed
             //  blocks.
+            let epoch = SortitionDB::get_stacks_epoch(tx, self.block_height)?.expect(&format!(
+                "FATAL: impossible block height: no epoch defined for {}",
+                self.block_height
+            ));
             let miss_distance = if actual_modulus > intended_modulus {
                 actual_modulus - intended_modulus
             } else {
                 BURN_BLOCK_MINED_AT_MODULUS + actual_modulus - intended_modulus
             };
-            if miss_distance > self.block_height {
-                return Err(op_error::BlockCommitBadModulus);
-            }
-            let intended_sortition = tx
-                .get_ancestor_block_hash(self.block_height - miss_distance, &tx_tip)?
-                .ok_or_else(|| op_error::BlockCommitNoParent)?;
+            let intended_sortition = match epoch.epoch_id {
+                StacksEpochId::Epoch21 => {
+                    // correct behavior
+                    if self.block_height < burnchain.first_block_height {
+                        return Err(op_error::BlockCommitPredatesGenesis);
+                    }
+                    let sortition_height = self
+                        .block_height
+                        .saturating_sub(burnchain.first_block_height);
```

Comment:
```suggestion
                    let sortition_height = self
                        .block_height
                        .checked_sub(burnchain.first_block_height)
                        .ok_or_else(|| op_error::BlockCommitPredatesGenesis)?;
```

Using `checked_sub` is more idiomatic than a manual check + saturating sub.